addons/mail/static/src/core/common/thread.js

```diff
@@ -155,6 +189,9 @@ export class Thread extends Component {
                 this.lastJumpPresent = nextProps.jumpPresent;
             }
             this.threadService.fetchNewMessages(nextProps.thread);
+            if (!this.env.messageHighlight.highlightedMessageId) {
+                this.threadService.fetchNewMessages(this.props.thread);
+            }
```

Comment:
I don't have any idea with it's relevant for it having and highlighted message to fetch new messages.
This looks wrong. If there's a bug and this fixes it, certainly this is not the correct fix. What was the problem you had without this?